classes/blocks/class-counter.php

```diff
@@ -105,7 +105,7 @@ public function prepare_data( $fields ) : array {
 		}
 		$target = floatval( $fields['target'] );
 
-		if ( array_key_exists( 'completed_api', $fields ) ) {
+		if ( array_key_exists( 'completed_api', $fields ) && $fields['completed_api'] ) {
```

Comment:
I've added a similar change to conversions branch.
You could check  `! empty( $fields['completed_api'])` which will check both if the key exists and it does not have an empty value